### DISCUSSION TITLE
Change pin mode pi4j v2.0

### DISCUSSION THREAD

**Discussion by RudNick91** [Pi4J/pi4j#76] [Q&A] [answered]
How to change pin mode in v2.0 ?

Im creating output pin like that :

```
     var pin17out= DigitalOutput.newConfigBuilder(pi4j)
                .id("pin17")
                .address(17)
                .shutdown(DigitalState.HIGH)
                .initial(DigitalState.HIGH)
                .provider("raspberrypi-digital-output");
                
        var pinOut= pi4j.create(pin17out);
```
        
And now I wanted to do : 

```
        var pin17in= DigitalInput.newConfigBuilder(pi4j)
                .id("pin17")
                .address(17)
                .provider("raspberrypi-digital-input");

        var pinIn= pi4j.create(pin17in);
```
        
Will it work?
I now have 2 pins on the same port .. shouldn't I kill one somehow?

**Comment by savageautomate** [comment, marked as answer]
@RudNick91 

I started a pull request for the `MultiPurposeDigitalInputOutput` here: https://github.com/Pi4J/pi4j-v2/pull/30

Here is the original issue tracking this change: https://github.com/Pi4J/pi4j-v2/issues/26

However, I have not had the time to fully test it and get it merged into the project yet.  Its something I cam probably start digging back into over the next couple of weeks.

I think this would address the issue you have raised.  You would create a single IO object and be able to call the `mode()` method to change to the desired INPUT or OUTPUT mode as needed.

**Reply by RudNick91** [reply]
Thank you for the quick reply.

Yes I have the same problem then #26.
I am trying to rewrite the code from v1.4 to handle DHT11.
DHT11 receives and sends data via the same pin. 

Can I build and work with version from branch issue/#26 locally without tests?
Or some other workaround for now?

**Comment by savageautomate** [comment]
Yes, you can try to build from this branch `https://github.com/Pi4J/pi4j-v2/tree/issue/%2326` and see what happens.  
I'm really not sure if this feature was completed or not.  But trying it out is a good way to find out :-) 

That branch won't contain some of the newer changes, but should be good enough to test with. 

Thanks, Robert

**Comment by relaxmax01** [comment]
rather old thread, but very interrested by the outcome: 

is this functionality (MultiPurposeDigitalInputOutput) available in Pi4J 2.2.1 and is there any doc or example about setting up a GPIO as dual purpose?
I'm trying to find the best way to monitor the state of an input and in some occasion be able to write a Digital.high or low to it.

Thank you!
